Auto Dashboard releases

SimonFischer04 opened the pull request:
Fixes #118

SimonFischer04 commented:
This should work after we get the change in the release-script. Now working on adding the github-action.

DutchmanNL commented:
> This should work after we get the change in the release-script. Now working on adding the github-action.

hmm, its failing for me at my local desk


```
> iobroker.esphome@0.3.1 dashboardRelease
> ts-node scripts/dashboard-release.mts

(node:68745) ExperimentalWarning: Importing JSON modules is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
file:///Users/nl69zh/Developer/community/opensource/ioBroker.esphome/scripts/dashboard-release.mts:42
    const response = await fetch('https://api.github.com/repos/esphome/esphome/releases')
                     ^
ReferenceError: fetch is not defined
    at getLatestDashboardVersion (file:///Users/nl69zh/Developer/community/opensource/ioBroker.esphome/scripts/dashboard-release.mts:42:22)
    at run (file:///Users/nl69zh/Developer/community/opensource/ioBroker.esphome/scripts/dashboard-release.mts:15:50)
    at file:///Users/nl69zh/Developer/community/opensource/ioBroker.esphome/scripts/dashboard-release.mts:47:7
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:530:24)
    at async loadESM (node:internal/process/esm_loader:91:5)
    at async handleMainPromise (node:internal/modules/run_main:65:12)

```

DutchmanNL commented:
> This should work after we get the change in the release-script.

what change do we need? script itself runs great nice job!

SimonFischer04 commented:
> what change do we need? script itself runs great nice job!\n\nedit: Understood, related to AlCalzone/release-script#154

Yes and no. Only seems to run fine. Because when you look at the script you can see that the actuall running of the release script is commented out for now so that i can test the rest.
And you already found the release script pr..

If you arent in a rush to add this feature instantly i would for now jsut leave this as it is and wait a bit to see if upstream gets merged,

 otherwiese i could temporarily add some overidde for the upstream compile changelog-plugin.

DutchmanNL commented:
> If you arent in a rush to add this feature instantly i would for now jsut leave this as it is and wait a bit to see if upstream gets merged,

no rush I think the current concept and PR is the right way to go. Let's wait for feedback by maintainer of release script and follow-up.
Nice journey anyway, automated update/integration and suggestion for release script covered by an automated action.... like it

SimonFischer04 commented:
> hmm, its failing for me at my local desk\n\nEdit: solved by 76d2b15

I dont know if its necessary to introduce another dependency as node 16 is already EOL and its run only inside the github action anyway , so the end user of the adapter does not need node18.

DutchmanNL commented:
> > hmm, its failing for me at my local desk\n\nEdit: solved by 76d2b15
> 
> I dont know if its necessary to introduce another dependency as node 16 is already EOL and its run only inside the github action anyway , so the end user of the adapter does not need node18.

Only temporary so I am also able to run it locally until my it department finally deploys 18+ 😅

SimonFischer04 commented:
> Nice journey anyway, automated update/integration and suggestion for release script covered by an automated action.... like it

yeah I may kind of tend to "overengineer" my contributions - sometimes simple bugs led me to things that i think can be improved and then it may escalate a bit :D.
for the esphome adapter my original issue was that I updated to debian 12 => new python version => broke the dashboard. and then, well stuff happened... :smile: .
 Also always trying to get fixes upstreamed to help others (release script, actually also fixed typescript types for the client class in the nativeapi - dependency still needs to be updated here), ....

Had I similar journey a few months ago where another project had a dependency that updated which broke a template because the dependency changed from a default export to a named one.
So basically all I would have needed to do was change something like `import func from 'dep'` to `import { func } from 'dep'` in my code generated from the template. and well. ... I ended up fixing some stuff end re-writing the whole project generator to be in es-module format ...

SimonFischer04 commented:
> Only temporary so I am also able to run it locally until my it department finally deploys 18+ 😅

Well... Good that I have root / sudo access to all of my devices :smile:

SimonFischer04 commented:
> > If you arent in a rush to add this feature instantly i would for now jsut leave this as it is and wait a bit to see if upstream gets merged,
> 
> no rush I think the current concept and PR is the right way to go. Let's wait for feedback by maintainer of release script and follow-up. Nice journey anyway, automated update/integration and suggestion for release script covered by an automated action.... like it

@DutchmanNL 
Does the new implementation in the adapter directlly: https://github.com/DrozmotiX/ioBroker.esphome/issues/118#issuecomment-1837538091 mean this is now useless?

DutchmanNL commented:
> @DutchmanNL Does the new implementation in the adapter directlly: [#118 (comment)](https://github.com/DrozmotiX/ioBroker.esphome/issues/118#issuecomment-1837538091) mean this is now useless?

Reviewing all PR’s it indeed feels now this is obsolete.

idea behind it and concept was great and also for me a interesting exercise to explore what’s possible with GitHub actions.

but as we were able to handle the proper version management now completely in iObroker and the Adapter configuration itself this should not be needed anymore and we can simplify the maintenance.

do you agree @SimonFischer04 ?

SimonFischer04 commented:
> > @DutchmanNL Does the new implementation in the adapter directlly: [#118 (comment)](https://github.com/DrozmotiX/ioBroker.esphome/issues/118#issuecomment-1837538091) mean this is now useless?
> 
> Reviewing all PR’s it indeed feels now this is obsolete.
> 
> idea behind it and concept was great and also for me a interesting exercise to explore what’s possible with GitHub actions.
> 
> but as we were able to handle the proper version management now completely in iObroker and the Adapter configuration itself this should not be needed anymore and we can simplify the maintenance.
> 
> do you agree @SimonFischer04 ?

@DutchmanNL 

Not really anymore.

Still getting issues / forum messages every few weeks about conflicting python / pillow / dashboard versions.
I don;t know if it's really the best idea to leave this up to the default user ...

"New" idea:
Add a new "adapter included (recommended)" default option to the dashboard selecting,  that will be handled by the automation logic in this pr (nicely fits together with newly introduced adapter tests). (combined with the (recently) instruduced auto adapter updates in admin).
Keep the other options for advanced users or debugging.

With this, we should hopefully finally rule out the recurring version mismatch issues (because it is automatically verified to work by the github action) for the average adapter user.
And still allow advanced user customization.


what do you think about that?

DutchmanNL commented:
> what do you think about that

I like that proposal!
So instead of always latest we predict the best which is a good idea

Still wondering if this will solve the installation/start issues sometimes but we will have to figure out

Are going to have doe that as a new PR/proposal, how would you like to proceed ?
